openmdao/docs/features/recording/basic_recording.rst

```diff
@@ -0,0 +1,205 @@
+change the amount of information retained by the recorders.
+
+A basic example of how to set an option:
+
+.. code-block:: console
+
+    self.my_recorder.options['record_desvars'] = True
+
+
+General Recording Options
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    options['record_metadata'] :  bool(True)
+        Tells recorder whether to record variable attribute metadata.
+    options['includes'] :  list of strings("*")
+        Patterns for variables to include in recording across all objects.
+    options['excludes'] :  list of strings('')
+        Patterns for variables to exclude in recording across all objects (processed after includes).
+
+System Recording Options
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+    options['record_outputs'] :  bool(True)
+        Tells recorder whether to record the outputs of a System.
+    options['record_inputs'] :  bool(False)
+        Tells recorder whether to record the inputs of a System.
+    options['record_residuals'] :  bool(False)
+        Tells recorder whether to record the residuals of a System.
+
+Driver Recording Options
+^^^^^^^^^^^^^^^^^^^^^^^^
+    options['record_desvars'] :  bool(True)
+        Tells recorder whether to record the desvars of a Driver.
+    options['record_responses'] :  bool(False)
+        Tells recorder whether to record the responses of a Driver.
+    options['record_objectives'] :  bool(False)
+        Tells recorder whether to record the objectives of a Driver.
+    options['record_constraints'] :  bool(False)
+        Tells recorder whether to record the constraints of a Driver.
+
+Solver Recording Options
+^^^^^^^^^^^^^^^^^^^^^^^^
+    options['record_abs_error'] :  bool(True)
+        Tells recorder whether to record the absolute error of a Solver.
+    options['record_rel_error'] :  bool(True)
+        Tells recorder whether to record the relative error of a Solver.
+    options['record_solver_output'] :  bool(False)
+        Tells recorder whether to record the output of a Solver.
+    options['record_solver_residuals'] :  bool(False)
+        Tells recorder whether to record the derivatives of a Solver.
+
+
+How To Attach a Recorder to an Object
++++++++++++++++++++++++++++++++++++++
+
+So you have a recorder created, and you've set the options you'd like.  Next, you need to attach the recorder to an
+object or objects using the `add_recorder` command.
+
+Here's an example of adding a recorder to the top-level problem's driver:
+
+.. code-block:: console
+
+    self.prob.driver.add_recorder(self.my_recorder)
+
+A recorder can be attached to more than one object.  Also, more than one recorder can be attached to an object.
+
+
+A More Comprehensive Example
+++++++++++++++++++++++++++++
+
+.. code-block:: console
+
+    def test_simple_driver_recording(self):
+        if OPT is None:
+            raise unittest.SkipTest("pyoptsparse is not installed")
+
+        if OPTIMIZER is None:
+            raise unittest.SkipTest("pyoptsparse is not providing SNOPT or SLSQP")
```

Comment:
Yes...currently, it's failing because it has to create a database and query it during the docbuild, I figured I'd put in a story to make that work later.